presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchClient.java

```diff
@@ -102,24 +121,47 @@ public ElasticsearchClient(ElasticsearchConfig config)
 
         this.scrollSize = config.getScrollSize();
         this.scrollTimeout = config.getScrollTimeout();
+        this.refreshInterval = config.getNodeRefreshInterval();
+        this.tlsEnabled = config.isTlsEnabled();
+    }
 
-        // discover other nodes in the cluster and add them to the client
-        // TODO: refresh periodically
-        HttpHost[] hosts = getNodes().values().stream()
-                .map(Node::getAddress)
-                .map(address -> HttpHost.create(format("%s://%s", config.isTlsEnabled() ? "https" : "http", address)))
-                .toArray(HttpHost[]::new);
-
-        client.getLowLevelClient().setHosts(hosts);
+    @PostConstruct
+    public void initialize()
+    {
+        if (!started.getAndSet(true)) {
+            executor.scheduleWithFixedDelay(this::refreshNodes, 0, refreshInterval.toMillis(), TimeUnit.MILLISECONDS);
```

Comment:
Typically, I run the refresh once manually to ensure we have an initial value.  With this change the nodes could be empty when the first query comes in.

Reply:
I'll add it, bu I'm not sure it matters in this case. The initial delay is 0, and the server won't report "started" until the connectors finish loading. It's unlikely a query is going to win that race.